src/include/duckdb/common/types/string_type.hpp

```diff
@@ -97,6 +97,10 @@ struct string_t {
 		return value.inlined.length;
 	}
 
+	void SetSize(uint32_t size) {
```

Comment:
I don't know if I like having a method for this.
If this is used incorrectly it will cause problems down the line.

For example if it was used to create a non-inlined string and someone thinks: "Ah I can make a substring out of this very easily with this" and the size would now indicate that it's inlined

Reply:
Perhaps we can add an `Unsafe` to it?

Reply:
Perhaps `SetSizeAndFinalize`, which does both, and then also calls `VerifyCharacters()`?

Reply:
> We could also do this:
> 
> ```c++
> new (&result.str) string_t(UnsafeNumericCast<uint32_t>(decompressed_string_size));
> ```
> 
> To achieve exactly the same, that saves us from creating the method at all.

This looks scary, I like the `SetSizeAndFinalize` more